vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/pipeline_util.cc

```diff
@@ -226,6 +227,55 @@ absl::optional<PipelineInfo> ParsePipelineValue(base::Value resource_value) {
                       std::move(linear_model_optional.value()));
 }
 
+absl::optional<PipelineEmbeddingInfo> ParsePipelineEmbedding(
+    base::Value resource_value) {
+  if (!resource_value.is_dict()) {
+    return absl::nullopt;
+  }
+
+  absl::optional<int> version_value = resource_value.FindIntKey("version");
+  if (!version_value.has_value()) {
+    return absl::nullopt;
+  }
+  int version = version_value.value();
+
+  std::string* timestamp_value = resource_value.FindStringKey("timestamp");
+  if (!timestamp_value) {
+    return absl::nullopt;
+  }
+  std::string timestamp = *timestamp_value;
```

Comment:
Please rebase and check modern `base::Value::Dict`/`FindString` etc. as we should now use the recent base::Value changes